File: data-structures-algorithms/searching/two-crystal-balls/twoCrystalBall.ts
function twoCrystalBall(breaks: boolean[]): number {
  const jump = Math.floor(Math.sqrt(breaks.length));
  let i = jump;
  for (; i < breaks.length; i += jump) {
    if (breaks[i]) {
      break;
    }
  }
  i -= jump;
  for (let j = 0; j < jump; j++, i++) {
    if (breaks[i]) return i;
  }
  return -1;
}

console.log(
  twoCrystalBall([
    false,
    false,
    false,
    false,
    false,
    false,
    false,
    false,
    false,
    false,
    false,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
    true,
  ])
);

File: data-structures-algorithms/searching/two-crystal-balls/two_crystal_ball.py
import math

def two_crystal_balls(breaks:list[bool])->int:
    jump = math.floor(math.sqrt(len(breaks)))
    i = jump
    for i in range(jump,len(breaks),jump):
        if breaks[i]:
            break
    i -= jump
    for j in range(i,i+jump):
        if breaks[j]:
            return j
    return -1

print(two_crystal_balls([False,False,False,False,False,False,False,False,False,False,True]))